src/main/java/Duke.java

```diff
@@ -1,10 +1,121 @@
+                    break;
+                } else if (command.equals(commands.LIST.toString())) {
+                    System.out.println("Here are the tasks in your list");
+                    if (tasks.size() != 0) {
+                        for (int i = 0; i < tasks.size(); i++) {
+                            System.out.println((i + 1) + "." + tasks.get(i).toString());
+                        }
+                    }
+                } else if (command.equals(commands.MARK.toString())) {
+                    int indexOfTask = Integer.parseInt(remainingInput) - 1;
+                    Task currentTask = tasks.get(indexOfTask);
+                    currentTask.markAsDone();
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println("[" + currentTask.getStatusIcon() + "] " + currentTask.getDescription());
+                } else if (command.equals(commands.UNMARK.toString())) {
+                    int indexOfTask = Integer.parseInt(remainingInput) - 1;
+                    Task currentTask = tasks.get(indexOfTask);
+                    currentTask.markAsUndone();
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    System.out.println("[" + currentTask.getStatusIcon() + "] " + currentTask.getDescription());
+                } else if (command.equals(commands.TODO.toString())) {
+                    if (remainingInput == null) {
+                        throw new IncorrectInputException("Enter a description!");
+                    }
+                    String description = remainingInput;
+                    Task newTask = new Todo(description);
+                    tasks.add(newTask);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(newTask);
+                    System.out.println("Now you have " + tasks.size() + Duke.numOfTasks(tasks) + "in the list.");
+                } else if (command.equals(commands.DEADLINE.toString())) {
+                    if (remainingInput == null) {
+                        throw new IncorrectInputException("Enter a description with a deadline!");
+                    }
+                    String[] deadlineDetails = remainingInput.split(" /by ", 2);
+                    String deadlineDescription = deadlineDetails[0];
+                    String deadline = deadlineDetails.length > 0 ? deadlineDetails[1] : null;
+
+                    if (deadline == null) {
+                        throw new IncorrectInputException("Enter a valid deadline!");
+                    }
+                    Task newTask = new Deadline(deadlineDescription, deadline);
+                    tasks.add(newTask);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(newTask);
+                    System.out.println("Now you have " + tasks.size() + Duke.numOfTasks(tasks) + "in the list.");
```

Comment:
I like how the different logical units are separated by a line! The same goes for the rest of this if statement.